src/controllers/TalksController.php

```diff
@@ -140,8 +142,229 @@ public function postAction($request, $db) {
+        if (empty($talk['talk_description'])) {
+            $errors[] = "'description' is a required field";
+        }
+
+        $talk['language'] = filter_var($request->getParameter('language'), FILTER_SANITIZE_STRING);
+        if (empty($talk['language'])) {
+            $talk['language'] = 'English - UK';
+        }
+        if (! in_array($talk['language'], $talk_mapper->getLanguages())) {
+            $errors[] = sprintf('The language "%s" isn\'t known', $talk['language']);
+        }
+        $talk['slides_link'] = filter_var($request->getParameter('slides_link'), FILTER_SANITIZE_URL);
+
+        if (! $isAdmin) {
+            $talk['url_friendly_talk_title'] = $existing_talk['url_friendly_talk_title'];
+            $talk['type']               = $existing_talk['type'];
+            $talk['duration']           = $existing_talk['duration'];
+            $talk['start_date']               = (new \DateTime($existing_talk['start_date']))->format('U');
+        } else {
+
+            $talk['url_friendly_talk_title'] = filter_var(
+                $request->getParameter("url_friendly_talk_title"),
+                FILTER_SANITIZE_STRING
+            );
+            if (empty($talk['url_friendly_talk_title'])) {
+                $talk['url_friendly_talk_title'] = $talk['talk_title'];
+            }
+
+            $talk['type'] = filter_var(
+                $request->getParameter("type"),
+                FILTER_SANITIZE_STRING
+            );
+            if (empty($talk['type'])) {
+                $errors[] = "'type' is a required field";
+            }
+
+            if (! in_array($talk['type'], $talk_mapper->getCategories())) {
+                $errors[] = sprintf(
+                    'The given talk-category "%s" isn\'t recognized',
+                    $talk['type']
+                );
+            }
+
+            $talk['start_date'] = (new \DateTime($request->getParameter("start_date")))->format('U');
+
+            $talk['duration'] = filter_var(
+                $request->getParameter('duration'),
+                FILTER_SANITIZE_NUMBER_INT
+            );
+
+            if (empty($talk['duration'])) {
+                $talk['duration'] = 60;
+            }
+
+            $incoming_speakers_list = $request->getParameter('speakers');
+            if(is_array($incoming_speakers_list)) {
+                $talk['speakers'] = array_map(function($speaker){
+                    $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
+                    $speaker = trim($speaker);
+                    return $speaker;
+                }, $incoming_speakers_list);
+            }
         }
+
+        if ($errors) {
+            throw new Exception(implode(". ", $errors), 400);
+        }
+
+        $new_id = $talk_mapper->edit($talk, $talk_id);
+
+        $uri = $request->base . '/' . $request->version . '/talks/' . $new_id;
+        header("Location: " . $uri, true, 201);
```

Comment:
We haven't created the resource, so rather than a `201`, how about a `204` as per [RFC2616](http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.6) ?